.github/workflows/rust.yml

```diff
@@ -6,16 +6,38 @@ on:
     branches: [main]
 
 jobs:
-  rust-fmt-and-clippy:
+  rust-fmt:
     runs-on: UbuntuLatest32Cores128G
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
       - uses: DeterminateSystems/nix-installer-action@v4
       - uses: DeterminateSystems/magic-nix-cache-action@main
 
       - name: Rust formatting
         run: nix develop --command cargo fmt --check
 
+  rust-clippy-and-test:
+    # We need to run the checks on Linux and macOS because there are system-specific
+    # versions of fh (e.g. `fh apply nix-darwin` is only on macOS while `fh apply
+    # nixos` is only on Linux)
+    strategy:
```

Comment:
I don't think this is true anymore, so let's combine them back up and rename it back to `rust-fmt-and-clippy`

Reply:
Like just revert the chanegs to this file?

Reply:
Pretty much, except I imagine we'd want to keep the updated `actions/checkout`